packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2212,26 +2263,59 @@ function commitMutationEffectsOnFiber(
   const flags = finishedWork.flags;
 
   if (enableTransitionTracing) {
-    switch (finishedWork.tag) {
-      case HostRoot: {
-        const state = finishedWork.memoizedState;
-        const transitions = state.transitions;
-        if (transitions !== null) {
-          transitions.forEach(transition => {
-            // TODO(luna) Do we want to log TransitionStart in the startTransition callback instead?
-            addTransitionStartCallbackToPendingTransition({
-              transitionName: transition.name,
-              startTime: transition.startTime,
+    if (flags & SuspenseToggle) {
```

Comment:
We should be able to move all this to the passive phase, right?

Reply:
Actually we can't because the end time might be incorrect because the passive phase could be called after paint, which is why it lives in the sync commit phase

Reply:
But how would you know which time you should use?

Reply:
I don't understand the question. You would read the current time in the mutation phase, then pass that timestamp to the passive phase. Then all the logic after that is the same.